### PR TITLE
Creation of models

### LINKED ISSUE #17: Create Models In SQLAlchemy
As per the DB Schema (present in dropbox paper), we should make SQLAlchemy models.

The following guidelines to be followed:

- The current schema will be barebone and we will need to update them with constraints.
- Each model should be placed appropriately in the right file and module.

cc/ @mukul-mehta @grapheo12 Please coordinate on this :small_airplane: 

NOTE: Use proper docstrings with each class and method you write.

### PR DESCRIPTION
**Pull request by grapheo12**
Concerns #17
Created models for post and tag and the junction table connecting them.
Added flask_session, flask_bcrypt to dependencies.

### CODE REVIEW

File: app/main/models/enums.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+"""
+All enums used in the schema are described here. All models should refer
```
Comment (thealphadollar) [inline app/main/models/enums.py]: Good. We do not need to write "all models...." Just the first sentence is enough.

Even better is using, "The module contains all the Enums"

File: app/main/models/enums.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    startup
+    technology
+
+class CommentType(enum.Enum):
+    comment
+    report
+    highlight
+
+class PriorityType(enum.Enum):
+    less_of_these
```
Comment (thealphadollar) [inline app/main/models/enums.py]: Give each enum in this an integer as we discussed

File: app/main/models/posts.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+"""DB Model for the post table"""
+from . import db
+from enums import PostType
+from tags import tagPostJunction
+import datetime
+
+class Post(db.Model):
+    """Description of Post model.
```
Comment (thealphadollar) [inline app/main/models/posts.py]: The first line in a multiline docstring is empty.

```
"""
Description ...
"""
```

File: app/main/models/tags.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+"""DB Model for Tag and Junction Tables connecting to User and Post"""
+from . import db
+class Tag(db.Model):
+    """Description of Tag Model
```
Comment (thealphadollar) [inline app/main/models/tags.py]: same here.

**Comment by grapheo12** [comment]
@thealphadollar Please review. @mukul-mehta is working on the other models. Once everything is updated here. We can test the models.

**Comment by thealphadollar** [comment]
@grapheo12 Please resolve conflicts.

File: app/main/models/posts.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    #Relationships
+    author = db.relationship('User', backref='posts', lazy=False)
+    tags = db.relationship('Tag', secondary=postTagJunction, lazy='subquery',
+                            backref=db.backref('posts', lazy=True))
+    savers = db.relationship('User', secondary=postSaves, lazy=True,
+                            backref=db.backref('posts', lazy='subquery'))
+    images = db.relationship('ImgLink', secondary=imgPostJunction,
+                             lazy='subquery')
+    #TODO: Reactions and Ratings
```
Comment (thealphadollar) [inline app/main/models/posts.py]: Why is a TODO left?

Comment (grapheo12) [reply app/main/models/posts.py]: Oh Sorry! As per naming conventions, those junction tables ought to appear on users.py. I forgot to remove that.

**Pull request merged**